### PR TITLE
[node-barion] Add type definitions for node-barion

### PR DESCRIPTION
**Pull request by aron123**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by aron123** [comment]
@LaKing @polynar @n84ck @axos88 - If you are interested in using `node-barion` in a TypeScript environment seamlessly, and have enough time for it, please [review this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61048/files). Thanks for your help.

File: types/node-barion/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
+// Type definitions for node-barion 3.1
+// Project: https://github.com/aron123/node-barion
+// Definitions by: Áron Kiss <https://github.com/aron123>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
```
Comment (peterblazejewicz) [inline types/node-barion/index.d.ts:6]:
```suggestion
```
this is only required here, due to `console` methods being used in the tests. 
Those can be removed completely or replaced with proper type assertions:
https://github.com/DefinitelyTyped/DefinitelyTyped#my-package-teststs

File: types/node-barion/node-barion-tests.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    Amount: {
+        Currency: 'HUF' as 'HUF',
+        Value: 404
+    },
+    Comment: 'Buy me 1 EUR please.'
+};
+
+barion.emailTransfer(transfer, (err, data) => {
+    if (err && err.name === 'BarionModelError') {
+        console.error('BARION MODEL ERROR', err.message);
```
Comment (peterblazejewicz) [inline types/node-barion/node-barion-tests.ts:22]:
```suggestion
```

File: types/node-barion/node-barion-tests.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return;
+    } else if (err) {
+        console.error('OTHER ERROR', err);
```
Comment (peterblazejewicz) [inline types/node-barion/node-barion-tests.ts:25]:
```suggestion
```

File: types/node-barion/node-barion-tests.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return;
+    }
+
+    console.log(data);
```
Comment (peterblazejewicz) [inline types/node-barion/node-barion-tests.ts:29]:
```suggestion
```

File: types/node-barion/node-barion-tests.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+});
+
+async function sendOrder() {
+    try {
+        const data = await barion.emailTransfer(transfer);
+        console.log('DATA', data.arbitraryField);
```
Comment (peterblazejewicz) [inline types/node-barion/node-barion-tests.ts:35]:
```suggestion
```

File: types/node-barion/node-barion-tests.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    } catch (err) {
+        console.error('Some error occurred:', err);
```
Comment (peterblazejewicz) [inline types/node-barion/node-barion-tests.ts:37]:
```suggestion
```

File: types/node-barion/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/node-barion/index.d.ts:6]: word of explanation: each such dependency put  another dependency test on @types/node CI tests, which are run on any @types/node definition changes. More explicit dependencies, longer tests, etc. Your types do not use any of Node native symbols, requiring explicit dependency on @types/node, just language features.

**Comment by aron123** [comment]
@peterblazejewicz Thanks for the review, I applied your suggestions.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@aron123 thanks!

**Comment by aron123** [comment]
Ready to merge

**Pull request merged**